app/src/org/commcare/dalvik/odk/provider/InstanceProvider.java

```diff
@@ -222,6 +222,7 @@ public Uri insert(Uri uri, ContentValues initialValues) {
 
         SQLiteDatabase db = mDbHelper.getWritableDatabase();
         long rowId = db.insert(INSTANCES_TABLE_NAME, null, values);
+        db.close();
```

Comment:
as long as we're not using any db stuff between getting the database handle and using it.

Does getWritableDatabase cache its result? If so this might cause issues